extensions/network/configurationinternal.m

```diff
@@ -300,6 +300,97 @@ static int dynamicStoreLocalHostName(lua_State *L) {
     return 1 ;
 }
 
+// internal stuff to make setLocation work
+#define kSCPreferencesOptionChangeNetworkSet    CFSTR("change-network-set") // CFBooleanRef
+SCPreferencesRef
+SCPreferencesCreateWithOptions      (
+                                     CFAllocatorRef      allocator,
+                                     CFStringRef     name,
+                                     CFStringRef     prefsID,
+                                     AuthorizationRef    authorization,
+                                     CFDictionaryRef options
+                                     );
+
+/// hs.network.configuration:setLocation(location) -> boolean
+/// Method
+/// Switches to a new location
+///
+/// Parameters:
+///  * location - string containing UUID or name of new location
+///
+/// Returns:
+///  * bool
+static int dynamicStoreSetLocation(lua_State *L) {
+    LuaSkin *skin = [LuaSkin shared] ;
+    [skin checkArgs:LS_TUSERDATA, USERDATA_TAG, LS_TSTRING, LS_TBREAK];
+    
+    NSString *target ;
+    if (lua_gettop(L) == 2) {
+        target = [skin toNSObjectAtIndex:2];
+    }
+    AuthorizationRef authorization = NULL;
+    AuthorizationFlags flags = kAuthorizationFlagDefaults;
+    OSStatus status = AuthorizationCreate(NULL,
+                                          kAuthorizationEmptyEnvironment,
+                                          flags,
+                                          &authorization);
+
+    if (status != errAuthorizationSuccess) {
+        lua_pushboolean(L, 0);
```

Comment:
While I will be the first to admit that I haven't been consistent on this either, it kind of depends... my current thinking is that if a failure to authorize is a valid state that the user could get with this function/method, then it should return a value of false or nil (depending upon whichever makes the most sense in context); if it indicates a problem that means absolutely nothing can go forward, then code execution should halt right then and there so that the user knows where they need to fix or adjust something for this to be useful.

However, others may argue that it's up to the calling code to check for `false` and throw an error or do something else if they want to... and a counter to that is that if the calling code truly wants to handle an error condition itself, then it can always wrap the function/method that throws an error in `pcall`, so... it comes down to developer preference since we don't have a style guide yet.